src/discordtipbot/statistics.cr

```diff
@@ -0,0 +1,43 @@
+struct Statistics
+  DB.mapping({
+    transactions: {
+      type: Int64,
+      key:  "transaction_count",
+    },
+    total: {
+      type: BigDecimal,
+      key:  "transaction_sum",
+    },
+    tips: {
+      type: BigDecimal,
+      key:  "tip_sum",
+    },
+    soaks: {
+      type: BigDecimal,
+      key:  "soak_sum",
+    },
+    rains: {
+      type: BigDecimal,
+      key:  "rain_sum",
+    },
+  })
+
+  @@ttl : Time::Span = 1.minutes
+
+  @@last = Time.utc_now
+
+  def self.get(db : DB::Database)
+    update(db)
+    Statistics.from_rs(db.query("SELECT * FROM statistics")).last
```

Comment:
Doesn't `#query_one` just return a single `ResultSet` so you don't have to do `.last`? There will only ever be one row returned with the view you set up.

Reply:
`#query_one` is expected to be invoked with a block. `#query` just returns a `ResultSet`